### PR TITLE
Fixed inconsistency in ApiUrls.cs

### PR DESCRIPTION
**Pull request by dampir**
As part of my work on #1120 I've done some improvements in ApiUrls.cs class. 

I've found that in order to support work with repo by their ID we should add 83 new methods in ApiUrls.cs that would duplicate methods that allow to work with repo by their owner and name identificators.

E.g.

We have the next method in ApiUrls.cs:

```cs
public static Uri RepoBranch(string owner, string name, string branchName)
{
      return "repos/{0}/{1}/branches/{2}".FormatUri(owner, name, branchName);
}
```

So we should add the new one:

```cs
public static Uri RepoBranch(int repositoryId, string branchName)
{
      return "repositories/{0}/branches/{1}".FormatUri(repositoryId, branchName);
}
```

Because of the fact that we should add 83 new methods, I've decided to generate them programmatically. And in order to get appropriate result of code generation, I've done some useful changes in ApiUrls.cs.

- [x]  Added XML documentation for needed methods.
- [x] Rename some method's parameters on order to eliminate conflicts of parameter's names.
- [x] Most of ApiUrls methods use "owner" and "name" as name of parameters that identify repository. But some of them use "repositoryName" or "repo" as parameter name. Now all of such methods use consistent parameter names: "owner" and "name".

BTW, I've implemented program that generate new 83 methods. Here is the [repo](https://github.com/dampir/OctokitUrlMethodsGenerator) and here is the [file](https://github.com/dampir/OctokitUrlMethodsGenerator/blob/master/new_methods.txt) with new generated methods. All methods contains appropriate XML documentation in many cases better than their original prototypes. I'm going to pull these methods in next PR after this one would be merged.

Now I'm thinking about how to validate them against their prototypes, how to check that each pair of methods returns equal results (e.g. RepoBranch(string owner, string name, string branchName) and RepoBranch(int repositoryId, string branchName) should return an equal JSONs if repositoryId is according to owner and name of repository).

/cc @shiftkey , @ryangribble

### CODE REVIEW

File: Octokit/Helpers/ApiUrls.cs [hunk 1]
```diff
@@ -556,60 +556,60 @@ public static Uri Milestones(string owner, string name)
         /// Returns the <see cref="Uri"/> that returns the specified label.
         /// </summary>
         /// <param name="owner">The owner of the repository</param>
-        /// <param name="repo">The name of the repository</param>
-        /// <param name="name">The milestone number</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="milestoneNumber">The milestone number</param>
```
Comment (ryangribble) [inline Octokit/Helpers/ApiUrls.cs]: this should be labelName right?

Comment (dampir) [inline Octokit/Helpers/ApiUrls.cs]: Yes, sure. Fixed!

Comment (shiftkey) [inline Octokit/Helpers/ApiUrls.cs]: @dampir was the commit pushed?

Comment (dampir) [inline Octokit/Helpers/ApiUrls.cs]: @shiftkey, yes, now parameter has name "labelName" in Label method.Here is commit 		0da6f0e

Comment (shiftkey) [inline Octokit/Helpers/ApiUrls.cs]: @dampir apologies, I didn't see the comments being folded up so I assumed it was still to-be-done

Comment (dampir) [inline Octokit/Helpers/ApiUrls.cs]: @shiftkey no problem! :smile:

**Comment by dampir** [comment]
Travis-ci falls again, please rerun it whenever possible.

**Pull request merged**